### PR TITLE
[8.11] PLANNER-2437 Use Quarkus 2 platform BOM

### PR DESCRIPTION
**Pull request by rsynek**
As previously agreed, this change targets only the `8.11.x` branch, while the `development` branch remains untouched.

### JIRA
https://issues.redhat.com/browse/PLANNER-2437

### CODE REVIEW

**Comment by antncl** [comment]
Looks good to me, I would maybe add a section into the readme that describes how to override this, like here:
https://github.com/kiegroup/kogito-examples/tree/main#use-alternative-quarkus-platforms

**Comment by rsynek** [comment]
> Looks good to me, I would maybe add a section into the readme that describes how to override this, like here:
> https://github.com/kiegroup/kogito-examples/tree/main#use-alternative-quarkus-platforms

Looking at the text in `kogito-examples`, I don't think it gives the best example. Although the `quarkus-universe-bom` is still part of the platform, it's recommended to use (multiple, if needed) fine-grained BOMs like described in https://quarkus.io/blog/quarkus-2x-platform-quarkiverse-registry/ .

File: technology/java-activemq-quarkus/pom.xml [hunk 1]
```diff
@@ -28,18 +30,16 @@
   <dependencyManagement>
     <dependencies>
       <dependency>
-        <groupId>io.quarkus</groupId>
-        <!-- Alternatively, use <artifactId>quarkus-universe-bom</artifactId>
-         which includes both quarkus-bom and optaplanner-bom. -->
-        <artifactId>quarkus-bom</artifactId>
-        <version>${version.io.quarkus}</version>
+        <groupId>${quarkus.platform.group-id}</groupId>
+        <artifactId>quarkus-optaplanner-bom</artifactId>
```
Comment (rsynek) [inline technology/java-activemq-quarkus/pom.xml:34]: @AntStephenson With the Quarkus 2.2.0.Final, the platform should already use OptaPlanner build parent as an input to generate this `quarkus-optaplanner-bom`. This artifact should define also versions of dependencies, so in theory, we should not need the core `quarkus-bom`.

Comment (antncl) [reply technology/java-activemq-quarkus/pom.xml:34]: @rsynek When I tried on Kogito Examples to build with just the quarkus-kogito-bom the build failed, I guess this changed with 2.2.0.Final? if that's true I guess that's OK. There is also evolving discussions on that ticket I linked you to for Kogito too about how it should work.

**Review by antncl — APPROVED** [review]

File: technology/java-activemq-quarkus/pom.xml (hunk 1 above)
Comment (rsynek) [reply technology/java-activemq-quarkus/pom.xml:34]: Not sure how far is Kogito with including the 3rd party dependencies into their BOM. Anyway, it should still work, just making the quarkus-bom unnecessary.

**Pull request merged**